launchpod/angular-launchpod/src/app/app.component.ts

```diff
@@ -1,19 +1,47 @@
-import {BreakpointObserver, Breakpoints} from '@angular/cdk/layout';
-import {Component} from '@angular/core';
-import {Observable} from 'rxjs';
-import {map, shareReplay} from 'rxjs/operators';
+import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
+import { Component, OnInit } from '@angular/core';
+import { LoginPopupComponent } from './login-popup/login-popup.component';
+import { FormHandlerService } from './form-handler.service';
+import { Observable } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
+
+export class AppComponent implements OnInit {
+
   title = 'angular-launchpod';
+  loginLink: string;
 
   isHandset$: Observable<boolean> =
       this.breakpointObserver.observe(Breakpoints.Handset)
           .pipe(map(result => result.matches), shareReplay());
 
-  constructor(private breakpointObserver: BreakpointObserver) {}
+  constructor(private breakpointObserver: BreakpointObserver, private formHandlerService: FormHandlerService, public dialog: MatDialog) {}
+
+  ngOnInit(): void {
+    this.getLoginStatus();
+  }
+
+  getLoginStatus() {
+    this.formHandlerService.getLoginData()
+      .subscribe((user) => {
+        if (!user.isLoggedIn) {
```

Comment:
nit: when you have an if {} else {}, it's clearer to always still with positive logic in the conditional - checking user.isLoggedIn and putting that code first would be better. otherwise reading this you'd have to understand that the else means !!user.isLoggedIn

Reply:
done.